### PR TITLE
aio: docs watch

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

File: aio/tools/transforms/authors-package/guide-package.js [hunk 1]
```diff
@@ -5,22 +5,37 @@
...
+const { basePackage, CONTENTS_PATH } = require('./base-package');
+const { readFileSync } = require('fs');
+const { resolve } = require('canonical-path');
+
+function createPackage(guideName) {
+
+  const guideFilePath = `${CONTENTS_PATH}/guide/${guideName}.md`;
+  const guideFile = readFileSync(guideFilePath, 'utf8');
+  const examples = [];
+  guideFile.replace(/<code-(?:pane|example) [^>]*path="([^"]+)"/g, (_, path) => examples.push('examples/' + path));
```
Comment (gkalpak) [inline aio/tools/transforms/authors-package/guide-package.js]: You probably don't want to use `replace` here.

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/tools/transforms/authors-package/guide-package.js (hunk 1 above)
Comment (petebacondarwin) [inline aio/tools/transforms/authors-package/guide-package.js]: Do you have a better suggestion?
I tried, `regex.exec` and `str.match()` but both were much more involved to simply get hold of the paths.

Comment (gkalpak) [inline aio/tools/transforms/authors-package/guide-package.js]: Sorry, I didn't realize you are not using the returned value. It's fine then.

**Review by gkalpak — APPROVED** [review]
Nice one 💯

File: aio/tools/transforms/authors-package/watchr.js [hunk 2]
```diff
@@ -22,10 +22,28 @@ function next(error) {
...
-console.log('Doc gen will run when you change a file in either of these folders.');
+let p = Promise.resolve();
 
-watchr.open(contentsPath, listener, next);
-watchr.open(apiPath, listener, next);
+if (process.argv.indexOf('--watch-only') === -1) {
+  console.log('===================================================================');
+  console.log('Running initial doc generation');
+  console.log('-------------------------------------------------------------------');
+  console.log('Skip the full doc-gen by running: `yarn docs-watch -- --watch-only`');
```
Comment (wardbell) [inline aio/tools/transforms/authors-package/watchr.js]: Thanks, Pete!

**Pull request closed without merging**